src/main/java/core/basesyntax/WorkWithFile.java

```diff
@@ -1,7 +1,58 @@
 package core.basesyntax;
 
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.File;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+
 public class WorkWithFile {
+    private int supply;
+    private int buy;
+    private int result;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        readFile(fromFileName);
+        writeToFile(toFileName);
+        supply = 0;
+        buy = 0;
+        result = 0;
+    }
+
+    private void readFile(String fileName) {
+        File file = new File(fileName);
+        try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
+            String string = reader.readLine();
+            while (string != null) {
+                String[] data = string.split(",");
+                String action = data[0];
+                String value = data[1];
+                if (action.equals("supply")) {
+                    supply += Integer.parseInt(value);
+                } else {
+                    buy += Integer.parseInt(value);
+                }
+                string = reader.readLine();
+            }
+            result = supply - buy;
+        } catch (IOException e) {
+            throw new RuntimeException("Cant read file " + fileName, e);
+        }
+    }
+
+    private String getReport() {
+        return "supply," + Integer.toString(supply).trim() + System.lineSeparator()
+                + "buy," + Integer.toString(buy).trim() + System.lineSeparator()
+                + "result," + Integer.toString(result).trim() + System.lineSeparator();
+    }
 
+    private void writeToFile(String fileName) {
+        File file = new File(fileName);
+        try (BufferedWriter writer = new BufferedWriter(new FileWriter(file, true))) {
```

Comment:
The `FileWriter` is opened in append mode (`true`), which means that each time `writeToFile` is called, the report will be appended to the existing content of the file. If the intention is to overwrite the file each time, consider removing the `true` parameter.